lifecycle.Rmd

```diff
@@ -7,52 +7,251 @@ status("drafting")
+    There are two reasons for this practice: First, the presence of a fourth component makes it easy to tell if you're dealing with a released or in-development version.
+    Also, the use of the fourth place means that you're not limited to what the next released version will be.
     `0.0.1`, `0.1.0`, and `1.0.0` are all greater than `0.0.0.9000`.
 
-    Increment the development version, e.g. from `9000` to `9001`, if you've added an important feature that another development package needs to depend on.
+    Increment the development version, e.g. from `9000` to `9001`, if you've added an important feature and you (or others) need to be able to detect or require the presence of this feature.
+    For example, this can happen when two packages are developing in tandem.
+    This is generally the only reason that we bother to increment the development version.
+    This makes in-development versions special and, in some sense, degenerate.
+    Since we don't increment the development component with each Git commit, the same package version number is associated with many different states of the package source, in between releases.
 
 The advice above is inspired in part by [Semantic Versioning](https://semver.org) and by the [X.Org](https://www.x.org/releases/X11R7.7/doc/xorg-docs/Versions.html) versioning schemes.
 Read them if you'd like to understand more about the standards of versioning used by many open source projects.
+But we should underscore that our practices our inspired by these schemes and are somewhat less regimented.
 Finally, know that other maintainers follow different philosophies on how to manage the package version number.
 
-The version number of your package increases with subsequent releases of a package, but it's more than just an incrementing counter -- the way the number changes with each release can convey information about what kind of changes are in the package.
-We discuss this and more in @sec-release-version.
-For now, just remember that the first version of your package should be `0.0.0.9000`.
-`usethis::create_package()` does this, by default.
-`usethis::use_version()` increments the package version; when called interactively, with no argument, it presents a helpful menu:
+## Major vs. minor vs. patch release
+
+The version number of your package is always increasing, but it's more than just an incrementing counter -- the way the number changes with each release can convey information about the nature of the changes.
+The transition from 0.3.1 to 0.3.2, which is a patch release, has a very different vibe from the transition from 0.3.2 to 1.0.0, which is a major release.
+A package version number can also convey information about where the package is in its lifecycle.
+For example, the version 1.0.0 often signals that the public interface of a package is considered stable.
+
+How do you decide which type of release to make, i.e. which component(s) of the version should you increment?
```

Comment:
Maybe it's worth distinguishing theoretical breaking vs actual breaking. Sometimes a theoretical break doesn't actually affect any code.

Reply:
Maybe it's worth specifically listing some breaking changes:

* Removing a function
* Remove an argument
* Decreasing the set of valid inputs to a function

And some things that generally aren't breaking changes:

* Adding a function (although there's some small chance on introducing a conflict in user code)
* Adding an argument
* Increasing the set of valid inputs.
* Changing the text of a print method or warning/error.

And then talk about things on the boundary, like fixing bugs?

Reply:
This has been incorporated now.